tasks/create_kvm.yml

```diff
@@ -2,32 +2,90 @@
 
 # Crear una KVM desde cero en el nodo Proxmox
 - name: Create a KVM from scratch in Proxmox node
-  proxmox_kvm:
-    node: "{{ node }}"
-    api_user: "{{ api_user }}"
-    api_host: "{{ api_host }}"
-    api_password: "{{ api_password }}"
-    description: "{{ description | default (omit) }}"
-    kvm: "{{ kvm | default (omit) }}"
-    ostype: "{{ ostype | default (omit) }}"
-    name: "{{ proxmox_hostname }}"
-    sockets: "{{ sockets | default (omit) }}"
-    cores: "{{ cores | default (omit) }}"
-    cpuunits: "{{ cpuunits | default (omit) }}"
-    cpulimit: "{{ cpulimit | default (omit) }}"
-    memory: "{{ memory | default (omit) }}"
-    timeout: "{{ pve_timeout }}"
-    balloon: "{{ balloon | default (omit) }}"
-    shares: "{{ shares | default (omit) }}"
-    vga: "{{ vga | default (omit) }}"
+  community.general.proxmox_kvm:
+    node: "{{ pve_node }}"
+    api_host: "{{ pve_api_host }}"
+    api_user: "{{ pve_api_user | default(omit) }}"
+    api_password: "{{ pve_api_password | default(omit) }}"
+    api_token_id: "{{ pve_api_token_id | default(omit) }}"
+    api_token_secret: "{{ pve_api_token_secret | default(omit) }}"
+    proxmox_default_behavior: "{{ pve_default_behavior | default('compatibility') }}"
+    validate_certs: "{{ pve_validate_certs | default(omit) }}"
+    name: "{{ pve_hostname }}"
+    vmid: "{{ pve_kvm_vmid | default(omit) }}"
+    timeout: "{{ pve_kvm_timeout | default(omit) }}"
+    description: "{{ pve_kvm_description | default(omit) }}"
+    kvm: "{{ pve_kvm_hardware_virtualization | default(omit) }}"
+    ostype: "{{ pve_kvm_ostype | default(omit) }}"
+    sockets: "{{ pve_kvm_sockets | default(omit) }}"
+    cores: "{{ pve_kvm_cores | default(omit) }}"
+    cpu: "{{ pve_kvm_cpu | default(omit) }}"
+    cpuunits: "{{ pve_kvm_cpuunits | default(omit) }}"
+    cpulimit: "{{ pve_kvm_cpulimit | default(omit) }}"
+    memory: "{{ pve_kvm_memory | default(omit) }}"
+    balloon: "{{ pve_kvm_balloon | default(omit) }}"
+    vga: "{{ pve_kvm_vga | default(omit) }}"
+    acpi: "{{ pve_kvm_acpi | default(omit) }}"
+    agent: "{{ pve_kvm_agent | default(omit) }}"
+    args: "{{ pve_kvm_args | default(omit) }}"
+    autostart: "{{ pve_kvm_autostart | default(omit) }}"
+    bios: "{{ pve_kvm_bios | default(omit) }}"
+    bootdisk: "{{ pve_kvm_bootdisk | default(omit) }}"
+    citype: "{{ pve_kvm_citype | default(omit) }}"
+    cicustom: "{{ pve_kvm_cicustom | default(omit) }}"
+    ciuser: "{{ pve_kvm_ciuser | default(omit) }}"
+    cipassword: "{{ pve_kvm_cipassword | default(omit) }}"
+    sshkeys: "{{ pve_kvm_sshkeys | default(omit) }}"
+    ipconfig: "{{ pve_kvm_ipconfig | default(omit) }}"
+    nameservers: "{{ pve_kvm_nameservers | default(omit) }}"
+    searchdomains: "{{ pve_kvm_searchdomains | default(omit) }}"
+    delete: "{{ pve_kvm_delete | default(omit) }}"
+    digest: "{{ pve_kvm_digest | default(omit) }}"
+    force: "{{ pve_kvm_force | default(omit) }}"
```

Comment:
They are not explicitly documented because thinking of not extending the `defaults/main.yml` too much with what would be a copy-paste of the module documentation, I preferred to define the `Most important/used vars` and `Other vars` sections, leaving in the latter links to the documentation and to the invocation of the module in `create_kvm.yml` to view all the parameters handled.

https://github.com/UdelaRInterior/ansible-role-proxmox-create-kvm/blob/acee43437d50598383cdda7a488b0a9ed6792895/defaults/main.yml#L215